// antd-pro-server/lib/rule.js
'use strict';

var _toConsumableArray2 = require('babel-runtime/helpers/toConsumableArray');

var _toConsumableArray3 = _interopRequireDefault(_toConsumableArray2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var _require = require('url'),
    parse = _require.parse;

// mock tableListDataSource


var tableListDataSource = [];
for (var i = 0; i < 46; i += 1) {
  tableListDataSource.push({
    key: i,
    disabled: i % 6 === 0,
    href: 'https://ant.design',
    avatar: ['https://gw.alipayobjects.com/zos/rmsportal/eeHMaZBwmTvLdIwMfBpg.png', 'https://gw.alipayobjects.com/zos/rmsportal/udxAbMEhpwthVVcjLXik.png'][i % 2],
    no: 'TradeCode ' + i,
    title: '\u4E00\u4E2A\u4EFB\u52A1\u540D\u79F0 ' + i,
    owner: '曲丽丽',
    description: '这是一段描述',
    callNo: Math.floor(Math.random() * 1000),
    status: Math.floor(Math.random() * 10) % 4,
    updatedAt: new Date('2017-07-' + (Math.floor(i / 2) + 1)),
    createdAt: new Date('2017-07-' + (Math.floor(i / 2) + 1)),
    progress: Math.ceil(Math.random() * 100)
  });
}

function getRule(req, res, u) {
  var url = u;
  if (!url || Object.prototype.toString.call(url) !== '[object String]') {
    url = req.url; // eslint-disable-line
  }

  var params = parse(url, true).query;

  var dataSource = [].concat((0, _toConsumableArray3.default)(tableListDataSource));

  if (params.sorter) {
    var s = params.sorter.split('_');
    dataSource = dataSource.sort(function (prev, next) {
      if (s[1] === 'descend') {
        return next[s[0]] - prev[s[0]];
      }
      return prev[s[0]] - next[s[0]];
    });
  }

  if (params.status) {
    var status = params.status.split(',');
    var filterDataSource = [];
    status.forEach(function (s) {
      filterDataSource = filterDataSource.concat([].concat((0, _toConsumableArray3.default)(dataSource)).filter(function (data) {
        return parseInt(data.status, 10) === parseInt(s[0], 10);
      }));
    });
    dataSource = filterDataSource;
  }

  if (params.no) {
    dataSource = dataSource.filter(function (data) {
      return data.no.indexOf(params.no) > -1;
    });
  }

  var pageSize = 10;
  if (params.pageSize) {
    pageSize = params.pageSize * 1;
  }

  var result = {
    list: dataSource,
    pagination: {
      total: dataSource.length,
      pageSize: pageSize,
      current: parseInt(params.currentPage, 10) || 1
    }
  };

  if (res && res.json) {
    res.json(result);
  } else {
    return result;
  }
}

function postRule(req, res, u, b) {
  var url = u;
  if (!url || Object.prototype.toString.call(url) !== '[object String]') {
    url = req.url; // eslint-disable-line
  }

  var body = b && b.body || req.body;
  var method = body.method,
      no = body.no,
      description = body.description;


  switch (method) {
    /* eslint no-case-declarations:0 */
    case 'delete':
      tableListDataSource = tableListDataSource.filter(function (item) {
        return no.indexOf(item.no) === -1;
      });
      break;
    case 'post':
      var _i = Math.ceil(Math.random() * 10000);
      tableListDataSource.unshift({
        key: _i,
        href: 'https://ant.design',
        avatar: ['https://gw.alipayobjects.com/zos/rmsportal/eeHMaZBwmTvLdIwMfBpg.png', 'https://gw.alipayobjects.com/zos/rmsportal/udxAbMEhpwthVVcjLXik.png'][_i % 2],
        no: 'TradeCode ' + _i,
        title: '\u4E00\u4E2A\u4EFB\u52A1\u540D\u79F0 ' + _i,
        owner: '曲丽丽',
        description: description,
        callNo: Math.floor(Math.random() * 1000),
        status: Math.floor(Math.random() * 10) % 2,
        updatedAt: new Date(),
        createdAt: new Date(),
        progress: Math.ceil(Math.random() * 100)
      });
      break;
    default:
      break;
  }

  var result = {
    list: tableListDataSource,
    pagination: {
      total: tableListDataSource.length
    }
  };

  if (res && res.json) {
    res.json(result);
  } else {
    return result;
  }
}

module.exports = {
  getRule: getRule,
  postRule: postRule
};
